internal/persistence/sql/migrations/uuidmapping/uuid_mapping_migrator.go

```diff
@@ -0,0 +1,190 @@
+import (
+	"database/sql"
+	"fmt"
+	"time"
+
+	"github.com/gobuffalo/pop/v6"
+	"github.com/gofrs/uuid"
+	"github.com/ory/x/popx"
+	"github.com/ory/x/sqlcon"
+)
+
+type (
+	// We copy the definitions of RelationTuple and UUIDMapping here so that the
+	// migration will always work on the same definitions.
+	RelationTuple struct {
+		// An ID field is required to make pop happy. The actual ID is a
+		// composite primary key.
+		ID                    uuid.UUID      `db:"shard_id"`
+		NetworkID             uuid.UUID      `db:"nid"`
+		NamespaceID           int32          `db:"namespace_id"`
+		Object                string         `db:"object"`
+		Relation              string         `db:"relation"`
+		SubjectID             sql.NullString `db:"subject_id"`
+		SubjectSetNamespaceID sql.NullInt32  `db:"subject_set_namespace_id"`
+		SubjectSetObject      sql.NullString `db:"subject_set_object"`
+		SubjectSetRelation    sql.NullString `db:"subject_set_relation"`
+		CommitTime            time.Time      `db:"commit_time"`
+	}
+	UUIDMapping struct {
+		ID                   uuid.UUID `db:"id"`
+		StringRepresentation string    `db:"string_representation"`
+	}
+	UUIDMappings []*UUIDMapping
+)
+
+func (RelationTuple) TableName() string { return "keto_relation_tuples" }
+func (UUIDMappings) TableName() string  { return "keto_uuid_mappings" }
+func (UUIDMapping) TableName() string   { return "keto_uuid_mappings" }
+
+var (
+	name       = "migrate-strings-to-uuids"
+	version    = "20220509000000000000"
+	Migrations = popx.Migrations{
+		{
+			Version:   version,
+			Name:      name,
+			Path:      name,
+			Direction: "up",
+			DBType:    "all",
+			Type:      "go",
+			Runner: func(_ popx.Migration, conn *pop.Connection, _ *pop.Tx) error {
+				for page := 1; ; page++ {
+					relationTuples, hasNext, err := getRelationTuples(conn, page)
+					if err != nil {
+						return fmt.Errorf("could not get relation tuples: %w", err)
+					}
+
+					for _, rt := range relationTuples {
+						if err := migrateSubjectID(conn, &rt); err != nil {
+							return fmt.Errorf("could not migrate subject ID: %w", err)
+						}
+						if err := migrateSubjectSetObject(conn, &rt); err != nil {
+							return fmt.Errorf("could not migrate subject set object: %w", err)
+						}
+						if err := migrateObject(conn, &rt); err != nil {
+							return fmt.Errorf("could not migrate object: %w", err)
+						}
+						if err := conn.Update(&rt); err != nil {
+							return fmt.Errorf("failed to update relation tuple: %w", err)
+						}
+					}
+
+					if !hasNext {
+						break
+					}
+				}
+
+				return nil
+			},
+		},
+		// We have to specify the "down" migration even if it is a no-op, since
+		// the migrator will still manipulate the version table in the database.
+		{
+			Version:   version,
+			Name:      name,
+			Path:      name,
+			Direction: "down",
+			DBType:    "all",
+			Type:      "go",
+			Runner: func(_ popx.Migration, _ *pop.Connection, _ *pop.Tx) error {
+				return nil
```

Comment:
Yes, I'd rather not. WDYT?

Reply:
Hm I mean it will completely break down migrations and make them basically useless... @aeneasr ?

Reply:
Ah, now that I think about this more I think it would be good to have a proper down migration. Because otherwise, the mapping table will just be deleted, and then there is no way to get the original strings from there.

Reply:
I don't think we should break down migrations. Having data loss on down migrations is acceptable in rare cases, when documented properly.